docs/lang/articles/faqs/faq.md

```diff
@@ -66,3 +66,112 @@ maintenance overheads.
 ### How do I program on less structured data structures (such as graphs and tetrahedral meshes) in Taichi?
 
 These structures have to be decomposed into 1D Taichi fields. For example, when representing a graph, you can allocate two fields, one for the vertices and the other for the edges. You can then traverse the elements using `for v in vertices` or `for v in range(n)`.
+
+### How to install Taichi on a server without Internet access?
+
+Follow these steps to install Taichi on a server without Internet access.
+
+1. From a computer with Internet access, download the installation packages for dill, sourceinspect, and taichi, ensuring that this computer has the same operating system as the target server:
+
+```plaintext
+pip download dill sourceinspect taichi
+```
+
+2. Copy all the downloaded `*.whl` packages to your local server.
+
+3. On your local server, change directory to **/home/<user_name>/<taichi_tmp>/** and install Taichi using the following command:
+
+```plaintext
+pip install ./*
```

Comment:
Install each downloaded *.whl packages. Note that you should first install all the dependencies and leave taichi installation to the last. 

You may use the following command to install a single *.whl package:
```
python -m pip install xxxx.whl
```